backend/connector-integration/src/connectors/helcim/transformers.rs

```diff
@@ -0,0 +1,904 @@
+    router_data_v2::RouterDataV2,
+    router_request_types::SyncRequestType,
+};
+use error_stack::ResultExt;
+use hyperswitch_masking::{PeekInterface, Secret};
+use serde::{Deserialize, Serialize};
+
+use crate::{connectors::helcim::HelcimRouterData, types::ResponseRouterData};
+
+// Note: HelcimRouterData is generated by the macro, so we don't define it here
+// The macro generates: pub struct crate::connectors::helcim::HelcimRouterData<RD: FlowTypes, T: PaymentMethodDataTypes + ..., T> { pub connector: Helcim<T>, pub router_data: RD }
+
+pub fn check_currency(
+    currency: common_enums::Currency,
+) -> Result<common_enums::Currency, errors::ConnectorError> {
+    if currency == common_enums::Currency::USD {
+        Ok(currency)
+    } else {
+        Err(errors::ConnectorError::NotSupported {
+            message: format!("currency {currency} is not supported for this merchant account"),
+            connector: "Helcim",
+        })?
+    }
+}
+
+// Auth Struct
+pub struct HelcimAuthType {
+    pub(super) api_key: Secret<String>,
+}
+
+impl TryFrom<&ConnectorAuthType> for HelcimAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
+                api_key: api_key.to_owned(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct HelcimPaymentsRequest<
+    T: PaymentMethodDataTypes
+        + std::fmt::Debug
+        + std::marker::Sync
+        + std::marker::Send
+        + 'static
+        + Serialize,
+> {
+    amount: common_utils::types::FloatMajorUnit,
+    currency: common_enums::Currency,
+    ip_address: Secret<String, pii::IpAddress>,
+    card_data: HelcimCard<T>,
+    invoice: HelcimInvoice,
+    billing_address: HelcimBillingAddress,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ecommerce: Option<bool>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct HelcimBillingAddress {
+    name: Secret<String>,
+    street1: Secret<String>,
+    postal_code: Secret<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    street2: Option<Secret<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    city: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    email: Option<common_utils::pii::Email>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct HelcimInvoice {
+    invoice_number: String,
+    line_items: Vec<HelcimLineItems>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct HelcimLineItems {
+    description: String,
+    quantity: u8,
+    price: common_utils::types::FloatMajorUnit,
+    total: common_utils::types::FloatMajorUnit,
```

Comment:
FloatMajorUnit is being used in multiple places so this can be imported at the starting and can be used directly.
```suggestion
    total: FloatMajorUnit,
```